crates/clvm-traits/src/clvm_encoder.rs

```diff
@@ -1,17 +1,33 @@
-use clvmr::{Allocator, NodePtr};
+use clvmr::{Allocator, Atom, NodePtr};
+use num_bigint::BigInt;
 
 use crate::{clvm_list, clvm_quote, ToClvm, ToClvmError};
 
 pub trait ClvmEncoder: Sized {
-    type Node: Clone + ToClvm<Self::Node>;
+    type Node: Clone + ToClvm<Self>;
 
-    fn encode_atom(&mut self, bytes: &[u8]) -> Result<Self::Node, ToClvmError>;
+    fn encode_atom(&mut self, atom: Atom<'_>) -> Result<Self::Node, ToClvmError>;
     fn encode_pair(
         &mut self,
         first: Self::Node,
         rest: Self::Node,
     ) -> Result<Self::Node, ToClvmError>;
 
+    fn encode_bigint(&mut self, number: BigInt) -> Result<Self::Node, ToClvmError> {
+        let bytes = number.to_signed_bytes_be();
+        let mut slice = bytes.as_slice();
+
+        // Remove leading zeros.
+        while !slice.is_empty() && slice[0] == 0 {
```

Comment:
this doesn't look right. If you want to encode a *positive* integer whose MSB is 1, you need one leading zero. .e.g. 255.

It would be good to have unit tests covering this

Reply:
There's a subsequent check `slice.len() > 1 && (slice[1] & 0x80 == 0x80)` which breaks the loop.

This code was taken directly from `clvm_rs` here:
https://github.com/Chia-Network/clvm_rs/blob/main/src/number.rs#L10